File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        if(arr.empty()) return 0;
        int n = arr.size();
        vector<int> maxArr(n,0);
        maxArr[0] = arr[0];
        for(int i = 1; i < n; ++i)
            maxArr[i] = max(maxArr[i-1], arr[i]);
        int count = 0;
        for(int i = 0; i < n; ++i)
            if(maxArr[i] == i) ++count;
        return count;
    }
};

File: 0769-max-chunks-to-make-sorted/0769-max-chunks-to-make-sorted.py
class Solution(object):
    def maxChunksToSorted(self, arr):
        if not arr: return 0
        for i in range(1, len(arr)):
            arr[i] = max(arr[i-1], arr[i])
        count = 0
        for i in range(len(arr)):
            if arr[i] == i:
                count += 1 
        return count
        